crates/nu-cli/src/types/deduction.rs

```diff
@@ -0,0 +1,1451 @@
+                                var,
+                                VarShapeDeduction::from_usage_with_alternatives(
+                                    &var.span,
+                                    &[SyntaxShape::Number, SyntaxShape::Unit],
+                                ),
+                            )?;
+                        } else {
+                            unreachable!("Only int or number or unit in binary with op = * allowed")
+                        }
+                    }
+                }
+                Operator::Divide => {
+                    if let Some(shape) = self.get_shape_of_binary_arg_or_insert_dependency(
+                        (var, expr),
+                        bin_spanned,
+                        (pipeline_idx, pipeline),
+                        registry,
+                    )? {
+                        match shape {
+                            SyntaxShape::Int | SyntaxShape::Number => {
+                                match var_side {
+                                    BinarySide::Left => {
+                                        //Binary is on left, number / int on right
+                                        self.checked_insert(
+                                            var,
+                                            VarShapeDeduction::from_usage_with_alternatives(
+                                                &var.span,
+                                                &[
+                                                    SyntaxShape::Number,
+                                                    SyntaxShape::Int,
+                                                    SyntaxShape::Unit,
+                                                ],
+                                            ),
+                                        )?;
+                                    }
+                                    BinarySide::Right => {
+                                        //TODO currently no unit type is supports 1/unit. This
+                                        //changes if there would be ever more complex unit types
+                                        //e.G. Frequency
+                                        self.checked_insert(
+                                            var,
+                                            VarShapeDeduction::from_usage_with_alternatives(
+                                                &var.span,
+                                                &[SyntaxShape::Number, SyntaxShape::Int],
+                                            ),
+                                        )?;
+                                    }
+                                }
+                            }
+                            SyntaxShape::Unit => {
+                                match var_side {
+                                    BinarySide::Left => {
+                                        //Must be unit / unit
```

Comment:
Do you mean the shape of division should be unit divided by unit? This might be too restrictive. Both of these are valid (or at least *should* be valid in the future):

```
12ms / 12ms
12ms / 12
```

Reply:
This case is 
```math
$var / unit 
```
$var should be unit here, as no unit currently supports a format of: `1 / unit`.

My take away of your comment is, that the current implementation is to hard and lengthy to grasp. I will refactor this part.